api/envoy/extensions/filters/http/jwt_authn/v3/config.proto

```diff
@@ -59,11 +59,19 @@ message JwtProvider {
 
   // Specify the `principal <https://tools.ietf.org/html/rfc7519#section-4.1.1>`_ that issued
   // the JWT, usually a URL or an email address.
+  // It is optional. If specified, it has to match the *iss* field in JWT.
+  // Its usage implications are following:
+  // * If a JWT has *iss* field and this field is specified, they have to match.
+  // * If a JWT doesn't have *iss* field,  not to check with this field.
```

Comment:
so envoy only checks that the `iss` field in JWT is matched with the `issuer` field here when they both exist, and skips the check in all other cases? can we rephrase a little bit, I find the current comment a bit hard to follow.

And regarding the JwtProvider lookup in line 68, isn't the `provider` defined in a map, how do you which one is the first one?

Reply:
Will rephase the comment to make it clear.

Good point.  "the first" is not deterministic with the proto map.  
Since "allow_missing" used with Jwt without "iss" are rare cases, I will not try to design a way to make "the first" deterministic.  I will just document it.  If anybody wants to use it,  they have to make sure there is not one provider without issuer specified.